### PR TITLE
EASY-2311: add email to Message for the Datamanager

### PR DESCRIPTION
**Pull request by jo-pol**
fixes EASY-2311 : add email to Message for the Datamanager

#### When applied it will
* [x] ~~check if the email href works in the emd when presented by the webui~~
* 
* 

#### How should this be manually tested?

* [x] see https://github.com/DANS-KNAW/easy-deposit-api/pull/181

#### related pull requests on github
to be merged with/after https://github.com/DANS-KNAW/easy-deposit-api/pull/181
which in turn is to be merged after https://github.com/DANS-KNAW/easy-schema/pull/76

### CODE REVIEW

File: src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala [hunk 1]
```diff
@@ -77,7 +77,7 @@ class EmdSpec extends FlatSpec with Matchers with Inside with CanConnectFixture
         emd.getEmdRights.getTermsLicense should contain only (new BasicString("http://opensource.org/licenses/MIT"), acceptBS)
         emd.getEmdOther.getEasRemarks should contain only(
           new BasicRemark("Message for the Datamanager: Beware!!! Very personal data!!!"),
-          new BasicRemark("Message for the Datamanager: according to the depositor user001 (First Namen) this dataset DOES contain Privacy Sensitive data.")
+          new BasicRemark("Message for the Datamanager: according to the depositor user001 (First Namen, <a href='mailto:does.not.exist@dans.knaw.nl>does.not.exist@dans.knaw.nl</a>) this dataset DOES contain Privacy Sensitive data.")
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala]: I have my doubt this will presented as a link.

Comment (rvanheest) [reply src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala]: Try it on deasy ;-)

Comment (jo-pol) [reply src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala]: Don't yet have a deasy up and running again. 
Tried it indirectly with a [deposit](https://teasy.dans.knaw.nl/deposit/deposit-form/c9350abf-66b3-416d-9ce5-03014428ede3) which is ingested as [easy-dataset:94](https://teasy.dans.knaw.nl/ui/datasets/id/easy-dataset:94/tab/1) the description field shows the html tag as plain text and thus confirms my doubts. However, the html of this indirecrt test is typed by the end user and I haven't checked yet whether the input and/or output is escaped.

Comment (jo-pol) [reply src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala]: as discussed: just email as plain text

File: src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala [hunk 2]
```diff
@@ -79,17 +79,28 @@ object EMD extends DebugEnhancedLogging {
   }
 
   private def addPrivacySensitiveRemark(emd: EasyMetadata, agreementsXml: Elem): Unit = {
-    val signerId = agreementsXml \ "depositAgreement" \ "signerId"
-    val easyAccount = (signerId \@ "easy-account").toOption
-    val fullname = signerId.text
+
+    def userNamePart: String = {
+      val signerId = agreementsXml \ "depositAgreement" \ "signerId"
+      (signerId.text, (signerId \@ "easy-account").toOption, (signerId \@ "email").toOption) match {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]: Please do an 'extract value' on the account and email and add `.text` to the `val signerId` for better readability

Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]: Also add a `toOption` to the `signerId`, as with the account and email. That way the pattern match is easier, I think

File: src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala [hunk 3]
```diff
@@ -79,17 +79,28 @@ object EMD extends DebugEnhancedLogging {
...
+        case (fullName, None, None) => fullName
+        case ("", None, Some(email)) => email
+        case ("", Some(account), None) => account
+        case ("", Some(account), Some(email)) => s"$account ($email)"
+        case (fullName, None, Some(email)) => s"$fullName ($email)"
+        case (fullName, Some(account), None) => s"$account ($fullName)"
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]: in the context of the text, I would suggest

```suggestion
        case (fullName, Some(account), None) => s"$fullName (username $account)"
```

File: src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala [hunk 4]
```diff
@@ -79,17 +79,28 @@ object EMD extends DebugEnhancedLogging {
...
+        case (fullName, Some(account), Some(email)) => s"$account ($fullName, $email)"
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]: in the context of the text, I would suggest

```suggestion
        case (fullName, Some(account), Some(email)) => s"$fullName (username $account, email $email)"
```

File: src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala [hunk 5]
```diff
@@ -79,17 +79,28 @@ object EMD extends DebugEnhancedLogging {
...
+    }
 
     val remark = Option((agreementsXml \\ "containsPrivacySensitiveData").text) match {
       case Some(boolText) =>
-        val privacySensitivePart = if (BooleanUtils.toBoolean(boolText)) "DOES"
-                                   else "DOES NOT"
-        val usernamePart = easyAccount.map(username => s"$username ($fullname)").getOrElse(fullname)
-
-        s"according to the depositor $usernamePart this dataset $privacySensitivePart contain Privacy Sensitive data."
+        s"according to the depositor $userNamePart this dataset ${ privacyPart(boolText) } contain Privacy Sensitive data."
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]:
```suggestion
        s"According to depositor $userNamePart, this dataset ${ privacyPart(boolText) } contain Privacy Sensitive data."
```

File: src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala [hunk 6]
```diff
@@ -79,17 +79,28 @@ object EMD extends DebugEnhancedLogging {
...
+        case (fullName, Some(account), Some(email)) => s"$account ($fullName, $email)"
+      }
+    }
+
+    def privacyPart(boolText: String): String = {
+      if (BooleanUtils.toBoolean(boolText)) "DOES"
+      else "DOES NOT"
+    }
 
     val remark = Option((agreementsXml \\ "containsPrivacySensitiveData").text) match {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]: I know it was already this way, but would you mind taking a look at:

1. shouldn't this be `(agreementsXml \\ "containsPrivacySensitiveData").text.toOption`
2. would it be easier to change the pattern match to
    ```scala
    .map(containsPrivacySensitiveData => ...)
    .getOrElse {
        logger.warn(...)
        "..."
    }
    ```

File: src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala [hunk 7]
```diff
@@ -58,7 +59,9 @@ class EmdSpec extends FlatSpec with Matchers with Inside with CanConnectFixture
     for (bag <- new File("src/test/resources/dataset-bags").listFiles()) {
       sdoSetDir.mkdirs()
       implicit val s: Settings = newSettings(bag)
-      EMD.create(sdoSetDir) shouldBe a[Success[_]]
+      (bag, EMD.create(sdoSetDir)) should matchPattern { // TODO use behave like
```
Comment (jo-pol) [inline src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala:62]: needed quick trouble shooting to figure out which bag failed. Refactoring to `behave like` is too far beyond the scope of the issue.

File: src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala [hunk 8]
```diff
@@ -79,22 +79,39 @@ object EMD extends DebugEnhancedLogging {
...
-      case Some(boolText) =>
-        val privacySensitivePart = if (BooleanUtils.toBoolean(boolText)) "DOES"
-                                   else "DOES NOT"
-        val usernamePart = easyAccount.map(username => s"$username ($fullname)").getOrElse(fullname)
-
-        s"according to the depositor $usernamePart this dataset $privacySensitivePart contain Privacy Sensitive data."
-      case None =>
-        logger.warn("The field containsPrivacySensitiveData could not be found in agreements.xml")
-        "it could not be determined if this dataset does contain Privacy Sensitive data."
+
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.stage/dataset/EMD.scala]:
```suggestion
```

File: src/test/scala/nl.knaw.dans.easy.stage/CanConnectFixture.scala [hunk 9]
```diff
@@ -30,8 +31,14 @@ trait CanConnectFixture {
           connection.connect()
           connection.disconnect()
           true
-        case _ => throw new Exception
+        case connection => throw new Exception ("expecting a HttpURLConnection but got " + connection)
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.stage/CanConnectFixture.scala]:
```suggestion
        case connection => throw new Exception("expecting a HttpURLConnection but got " + connection)
```

File: src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala [hunk 10]
```diff
@@ -74,14 +77,70 @@ class EmdSpec extends FlatSpec with Matchers with Inside with CanConnectFixture
...
+  }
+
+  it should "create a remark without a privacy claim" in {
+    easRemarksFromAgreements(replacing = "<containsPrivacySensitiveData>true</containsPrivacySensitiveData>", by = "") should
+      include("Message for the Datamanager: No statement by First Namen (user001, does.not.exist@dans.knaw.nl) could be found whether this dataset contains Privacy Sensitive data.")
+  }
+
+  it should "create a remark without a signer" in {
+    easRemarksFromAgreements(replacing = """<signerId easy-account="user001" email="does.not.exist@dans.knaw.nl">First Namen</signerId>""", by = "") should
+      include("Message for the Datamanager: No statement by  could be found whether this dataset contains Privacy Sensitive data.")
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.stage/lib/EmdSpec.scala]: 2 spaces... can you somehow improve on that?

**Review by rvanheest — APPROVED** [review]

**Pull request merged**